### PR TITLE
Make Vary always list-based

### LINKED ISSUE #272: fix grammar for "Vary" header field
it uses
   
    "*" / #vary

but "vary" is a token, and token allows "*" anyway. Also, see #7.

### PR DESCRIPTION
**Pull request by mnot**
Fixes #272. Leaves registration of "*" for #274.

### CODE REVIEW

File: draft-ietf-httpbis-semantics-latest.xml [hunk 1]
```diff
@@ -9368,25 +9368,25 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
...
+   with "*" indicating special handling.
 </t>
 <sourcecode type="abnf7230"><iref primary="true" item="Grammar" subitem="Vary"/>
-  <x:ref>Vary</x:ref> = "*" / 1#<x:ref>field-name</x:ref>
+  <x:ref>Vary</x:ref> = 1#( "*" / <x:ref>field-name</x:ref> )
 </sourcecode>
 <t>
-   A Vary field value of "*" signals that anything about the request might
+   A Vary field value is a list of field names that indicates
+   that the named request header fields, known as the selecting header fields,
```
Comment (reschke) [inline draft-ietf-httpbis-semantics-latest.xml:9379]: (pedantic): not necessarily "header", right?

File: draft-ietf-httpbis-semantics-latest.xml [hunk 2]
```diff
@@ -9368,25 +9368,25 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
    The "Vary" header field in a response describes what parts of a request
    message, aside from the method, Host header field, and target URI,
    might influence the origin server's process for selecting and representing
-   this response. The value consists of either a single asterisk ("*") or a
-   list of header field names (case-insensitive).
+   this response. The value consists a list of header field names (case-insensitive),
+   with "*" indicating special handling.
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:9372]:
```suggestion
   this response.
```

File: draft-ietf-httpbis-semantics-latest.xml [hunk 3]
```diff
@@ -9368,25 +9368,25 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
...
 <sourcecode type="abnf7230"><iref primary="true" item="Grammar" subitem="Vary"/>
-  <x:ref>Vary</x:ref> = "*" / 1#<x:ref>field-name</x:ref>
+  <x:ref>Vary</x:ref> = 1#( "*" / <x:ref>field-name</x:ref> )
 </sourcecode>
 <t>
-   A Vary field value of "*" signals that anything about the request might
+   A Vary field value is a list of field names that indicates
+   that the named request header fields, known as the selecting header fields,
+   might have a role in selecting the representation. The potential selecting
+   header fields are not limited to those defined by this specification.
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:9381]:
```suggestion
   A Vary field value is a list of request field names, known as the selecting header fields,
   that might have a role in selecting the representation for this response. Potential selecting
   header fields are not limited to those defined by this specification.
```

File: draft-ietf-httpbis-semantics-latest.xml [hunk 4]
```diff
@@ -12115,7 +12115,8 @@ Content-Encoding: gzip
 <x:ref>URI-reference</x:ref> = &lt;URI-reference, see <xref target="RFC3986" x:fmt="," x:sec="4.1"/>&gt;
 <x:ref>User-Agent</x:ref> = product *( RWS ( product / comment ) )
 
-<x:ref>Vary</x:ref> = "*" / ( field-name *( OWS "," OWS field-name ) )
+<x:ref>Vary</x:ref> = *( "," OWS ) ( "*" / field-name ) *( OWS "," [ OWS ( "*" /
+ field-name ) ] )
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:12119]: This bit is wrong -- is that because you haven't updated bap yet?

Comment (mnot) [reply draft-ietf-httpbis-semantics-latest.xml:12117]: very possibly; will try again.

Comment (mnot) [reply draft-ietf-httpbis-semantics-latest.xml:12117]: Hm, I re-made and it didn't change. @reschke ?

**Comment by reschke** [comment]
Yes, seems to be a bug in bap. Will check.

**Pull request merged**